compiler+runtime/src/cpp/jank/runtime/core/math.cpp

```diff
@@ -1702,4 +1702,40 @@ namespace jank::runtime
       },
       o);
   }
+
+  native_integer parse_long(object_ptr const o)
+  {
+    return visit_object(
+      [](auto const typed_o) -> native_real {
+        using T = typename decltype(typed_o)::value_type;
+
+        if constexpr(std::same_as<T, obj::persistent_string>)
+        {
+          return std::stol(typed_o->data);
+        }
+        else
+        {
+          throw erase(make_box("Expected string, got nil"));
```

Comment:
Is there a way to get the type name statically here?

Reply:
You have `T`, which is an object type. Every object type has a static `obj_type`, which is its enum value. From there, we have the constexpr `object_type_str` fn. So, this will get you a string of the object type at compile-time.

```cpp
object_type_str(T::obj_type)
```

Reply:
Good to know, although I opted for `object_type_str(o->type)` since we dropped `visit_object`.